project/src/store/properties-slice/properties-slice.ts

```diff
@@ -46,6 +46,25 @@ export const propertiesSlice = createSlice({
       })
       .addCase(fetchPropertiesNearbyAction.fulfilled, (state, action) => {
         state.propertiesNearby = action.payload;
+      })
+      .addCase(changeFavoriteStatusAction.fulfilled, (state, action) => {
+        for (const property of state.properties) {
+          if (property.id === action.payload.id) {
+            property.isFavorite = action.payload.isFavorite;
+            break;
+          }
+        }
+
+        for (const property of state.propertiesNearby) {
+          if (property.id === action.payload.id) {
+            property.isFavorite = action.payload.isFavorite;
+            break;
+          }
+        }
+
+        state.property = state.property?.id === action.payload.id ?
+          action.payload :
+          state.property;
```

Comment:
А здесь можно не обновлять property если она не активна
```suggestion
        if(state.property?.id === action.payload.id) {
          state.property = action.payload :
       }
```